app/assets/javascripts/application.js

```diff
@@ -338,6 +338,26 @@ $(function() {
   });
 });
 
+function check_for_unavailable_puppetclasses() {
+  var unavailable_classes = $('#puppet_klasses #selected_classes .unavailable');
```

Comment:
tbh: I think this needs to be rewritten in webpack, this way you can add tests to the function...